onboard/catkin_ws/src/controls/scripts/thruster_controls.py

```diff
@@ -99,63 +73,21 @@ def transform_twist(self, base_frame, target_frame, twist):
                          ang_local.vector.z])
 
     def update_thruster_allocs(self):
-        self.pid_outputs_local = self.transform_twist('odom', 'base_link', self.pid_outputs)
-        #rospy.loginfo(pid_outputs_local)
+        if self.enabled:
+            self.pid_outputs_local = self.transform_twist('odom', 'base_link', self.pid_outputs)
+
         for i in range(len(self.powers)):
-            if(self.powers[i]!=0):
-                self.pid_outputs_local[i]=self.powers[i]
+            if self.powers[i] != 0:
+                self.pid_outputs_local[i] = self.powers[i]
 
         self.t_allocs = self.tm.calc_t_allocs(self.pid_outputs_local)
 
-
-    def _on_x(self, x):
-        self.pid_outputs[0] = x.data
-        self.update_thruster_allocs()
-
-    def _on_y(self, y):
-        self.pid_outputs[1] = y.data
+    def _on(self, val, direction):
```

Comment:
on what? Give this method a more descriptive name

Reply:
clarified in 189a96e